nav2_bringup/launch/nav2_bringup_launch.py

```diff
@@ -1,64 +1,160 @@
-            package='nav2_amcl',
-            node_executable='amcl',
-            node_name='amcl',
-            output='screen',
-            parameters=[{ 'use_sim_time': use_sim_time}]),
-
-        launch_ros.actions.Node(
-            package='dwb_controller',
-            node_executable='dwb_controller',
-            output='screen',
-            parameters=[params_file]),
-
-        launch_ros.actions.Node(
-            package='nav2_navfn_planner',
-            node_executable='navfn_planner',
-            node_name='navfn_planner',
-            output='screen',
-            parameters=[{ 'use_sim_time': use_sim_time}]),
-
-        launch_ros.actions.Node(
-            package='nav2_simple_navigator',
-            node_executable='simple_navigator',
-            node_name='simple_navigator',
-            output='screen',
-            parameters=[{ 'use_sim_time': use_sim_time}]),
-
-        launch_ros.actions.Node(
-            package='nav2_mission_executor',
-            node_executable='mission_executor',
-            node_name='mission_executor',
-            output='screen',
-            parameters=[{ 'use_sim_time': use_sim_time}]),
-    ])
+    use_simulation = launch.substitutions.LaunchConfiguration('use_simulation')
+    simulator = launch.substitutions.LaunchConfiguration('simulator')
+    world = launch.substitutions.LaunchConfiguration('world')
+    params_file = launch.substitutions.LaunchConfiguration(
+        'params', default=[launch.substitutions.ThisLaunchFileDir(), '/nav2_params.yaml'])
+
+    declare_use_simulation_cmd = launch.actions.DeclareLaunchArgument(
+        'use_simulation', condition=IfCondition('True'),
+        default_value='True', description='Whether to run in simulation')
+
+    declare_simulator_cmd = launch.actions.DeclareLaunchArgument(
+        'simulator',
+        default_value='gzserver', description='The simulator to use (gazebo or gzserver)')
+
+    declare_world_cmd = launch.actions.DeclareLaunchArgument(
+        'world',
+        default_value=os.path.join(
+            get_package_share_directory('turtlebot3_gazebo'),
+            'worlds/turtlebot3_worlds/burger.model'),
+        description='Full path to world file to load')
+
+    declare_params_file_cmd = launch.actions.DeclareLaunchArgument(
+        'params_file',
+        description='Full path to the ROS2 parameters file to use for all launched nodes')
+
+    launch_dir = os.path.join(get_package_share_directory('nav2_bringup'), 'launch')
+    gz = launch.substitutions.LaunchConfiguration('gz', default=['gzserver'])
+
+    # Specify the actions
+
+    start_gazebo_cmd = launch.actions.ExecuteProcess(
+        condition=IfCondition(use_simulation),
+        cmd=[simulator, '-s', 'libgazebo_ros_init.so', world, ['__params:=', params_file]],
+        cwd=[launch_dir], output='screen')
+
+    start_robot_state_publisher_cmd = launch.actions.ExecuteProcess(
+        condition=IfCondition(use_simulation),
```

Comment:
We're going to want the robot state publisher if we're running on the real robot as well. You can probably get rid of the if use_simulation check for this one.

Reply:
Ok, fixed.